src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,109 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] arrayList;
+    private int size;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        arrayList[size] = value;
+        size++;
+        if (size == arrayList.length) {
+            growArrayIfItIsFull();
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        outOfBoundsCheckForMethodAddByIndex(index);
+        if (arrayList.length - size == 0) {
+            growArrayIfItIsFull();
+        }
```

Comment:
1. if you add reusage of add(T value) method you may get rid of second check method
2. move check of resize need to private method to make all public ones shorter and simpler
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        outOfBoundsCheck(index);
        growArrayIfItIsFull();
```